finding_the_source.Rmd

```diff
@@ -0,0 +1,83 @@
+# Finding the Source 
+
+This chapter discusses how you can have an overview of the R codebase. For instance, where to find the implmentation of a base function written in R and where to find a primitive implementation written C. You may want to find the source code of a function just out of curiosity or maybe to gain more insight into what a particular function is actually doing. Whatever be the case, reading the source code will help you to learn a lot about any function.
+
+## Finding R source code
+
+1. Find the R function with the code of interest. You will always be able to print the top-level function (or use `View(function_name)` in RStudio). Looking at the code for the body of this function will reveal what you need to do next:
+
+  * Can already see code of interest: stop here or skip to step 3 to find the corresponding file in the R sources.
+
+  * Code of interest is in nested R function: go to step 2.
+
+  * Top-level function is an S3 generic, identified by a call to `UseMethod()`. Use `methods(function_name)` to see available methods, then go to step 2.
+  
+  * Code of interest is in compiled code, identified by a call to `.C()`, `.Call()`, `.Fortran()`, `.External()`, or `.Internal()` and `.Primitive()`: go to section on compiled code.

```

Comment:
grammar: "`.Fortran()`, and `.External()`, or ...", otherwise "`.External()`, `.Internal()`, or `.Primitive()`"